### PR TITLE
New os Eltek WebPower - files + test files

### PR DESCRIPTION
**Pull request by sippe2**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

### CODE REVIEW

**Comment by sippe2** [comment]
I really hope that now I made everything right and my brainfarts are fixed. :)

**Comment by TheGreatDoc** [comment]
Does this works for you?

**Comment by sippe2** [comment]
Yes it works.

**Comment by TheGreatDoc** [comment]
How sure you are? I've just tested it and battery/rectifier data is not showed anywhere.
I mean for the sensors defined as **battery** and **rectifier**
Its true there is no error but they are just not discovered.

**Comment by sippe2** [comment]
Hmm... Just looking Eltek device at the moment and I have all values... Strange. Are you sure that you have all mibs in use?

**Comment by TheGreatDoc** [comment]
At least, the ones you have uploaded.
image

Thats what I've discovered.

**Comment by sippe2** [comment]
Looks identical for me... So I don't quite understand what wrong? :)

**Comment by TheGreatDoc** [comment]
This:
```
        battery:
            data:
                -
                    oid: batteryCapacityData
                    value: batteryTimeToDisconnect
                    num_oid: .1.3.6.1.4.1.12148.9.3.20.1.
                    index: 'batteryTimeToDisconnect.{{ $index }}'
                    descr: Battery time to disconnect
                -
                    oid: batteryCapacityData
                    value: batteryCapacityLeft    
                    num_oid: .1.3.6.1.4.1.12148.9.3.20.2.
                    index: 'batteryCapacityLeft.{{ $index }}'
                    descr: Battery capacity left
                -
                    oid: batteryCapacityData
                    value: batteryCapacityUsed    
                    num_oid: .1.3.6.1.4.1.12148.9.3.20.3.
                    index: 'batteryCapacityUsed.{{ $index }}'
                    descr: Battery capacity used
                -
                    oid: batteryCapacityData
                    value: batteryCapacityTotal   
                    num_oid: .1.3.6.1.4.1.12148.9.3.20.4.
                    index: 'batteryCapacityTotal.{{ $index }}'
                    descr: Battery capacity total
                -
                    oid: batteryCapacityData
                    value: batteryQuality         
                    num_oid: .1.3.6.1.4.1.12148.9.3.20.5.
                    index: 'batteryQuality.{{ $index }}'
                    descr: Battery quality
        rectifier:
            data:
                -
                    oid: rectifier
                    value: rectifierRectifiersActive
                    num_oid: .1.3.6.1.4.1.12148.9.5.2.
                    index: 'rectifierRectifiersActive.{{ $index }}'
                    descr: Number of active rectifiers
                -
                    oid: rectifier
                    value: rectifierTotalCurrent
                    num_oid: .1.3.6.1.4.1.12148.9.5.3.
                    index: 'rectifierTotalCurrent.{{ $index }}'
                    descr: Rectifiers total current output (Ampers)
                    divisor: 10
                -
                    oid: rectifier
                    value: rectifierUtilization
                    num_oid: .1.3.6.1.4.1.12148.9.5.4.
                    index: 'rectifierUtilization.{{ $index }}'
                    descr: Rectifiers current utilization
                -
                    oid: rectifier
                    value: rectifierInstalledRectifiers
                    num_oid: .1.3.6.1.4.1.12148.9.5.1.
                    state_name: rectifierInstalledRectifiers
                    descr: Number of installed rectifiers
```
This does nothing

**Comment by sippe2** [comment]
What you suggest?

**Comment by TheGreatDoc** [comment]
Just remove it ;)

**Comment by sippe2** [comment]
Tony! Actually you are right... There is one state case and others can perhaps place under under runtime. Thanks. :) (And finally I learned how to use git properly. Actually I tried to fix my fork but it was much easier to start all over. :) First I tried to use git desktop version and I really have to say that was very stupid and painful experiment. :) Much easier to use bash and git commands. Thanks for the help. Now I stand corrected I hope so.)

**Comment by murrant** [comment]
@sippe2 planning on updating the yaml?

**Comment by sippe2** [comment]
Yes sir. Unfortunately my work is hurting everything nice. :) Hopefully looks better now.

**Comment by sippe2** [comment]
Actually I had a little problem. Is there somewhere plans to add capacity value Ampere hour (Ah)? That's UPS and power source specific value needed to decribe runtime and capacity values.

**Comment by sippe2** [comment]
Looks that Travis-Ci hates me again and I don't have idea what I did wrong...

FAILURES!
Tests: 802, Assertions: 1872, Failures: 1.
Cleaning database...
The command "php scripts/pre-commit.php -u --db --snmpsim --fail-fast" exited with 1.

**Comment by murrant** [comment]
From the log:
`OS eltek-webpower: Discovered sensors data does not match that found in tests/data/eltek-webpower.json`

Since you changed your sensor definitions, you probably need to update the json data.

**Comment by sippe2** [comment]
Perhaps I need some tutorial for dummies how to get snmpsim to work. I really have difficulties to understand how to install it properly under librenms.

**Comment by laf** [comment]
> Perhaps I need some tutorial for dummies how to get snmpsim to work. I really have difficulties to understand how to install it properly under librenms.

Don't worry about that for now. Get your other files right and we can sort that out.

**Comment by sippe2** [comment]
I think this os+discovery is ready now. If there is something bad, I'll try to fix it. :)

File: includes/definitions/discovery/eltek-webpower.yaml [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+                    state_name: batteryBreakerStatus.{{ $index }}
+                    states:
+                        - { value: 0, generic: 0, graph: 0, descr: normal }
+                        - { value: 1, generic: 2, graph: 0, descr: Alarm }
+                -
+                    oid: rectifier
+                    value: rectifierRectifiersActive
+                    num_oid: .1.3.6.1.4.1.12148.9.5.2.
+                    state_name: rectifierRectifiersActive
+                    descr:
```
[the remark this answers is not in the document]
Comment (murrant) [reply includes/definitions/discovery/eltek-webpower.yaml]: The concept of tables is unknown to some vendors.

**Comment by murrant** [comment]
The concept of tables is unknown to some vendors.

**Comment by sippe2** [comment]
After couple days of thinking I don't have any clever way to solve this right. (Iaf asked some kind of sorting.) Perhaps I need to ask some help because this is little bit too much for my brains and low level coding skills. Like you saw the oid tree is not increasing normal way so we are also forced to use other queries than bulk walk. Quite popular and used teleoperator device but I have to say that the snmp side is not so nice at all. I wondered this is easy and nice device to start with librenms. I was wrong. :)

**Pull request merged**